# Statistical Analysis

### Loading our cleaned data

In [27]:
from load_cleaned_data import load_cleaned_data
import polars as pl
import py_stats as ps

df = load_cleaned_data()

## Descriptive Analysis

In [24]:
def descriptive_analytics(data):
    # Calculate measures of central tendency
    mean = ps.arithmetic_mean(data)
    median = ps.median(data)
    mode = ps.mode(data)
    # Calculate measures of dispersion
    variance = ps.variance(data,population=False)
    std_dev = ps.standard_deviation(data, population=False)
    range = ps.midrange(data)
    context = {
        "mean": mean,
        "median": median,
        "mode": mode,
        "variance": variance,
        "std_dev": std_dev,
        "range": range  
    }

    return context  

In [ ]:
# Run descriptive analytics for all numerical columns and display nicely
results = {}
numeric_cols = [col for col, dtype in zip(df.columns, df.dtypes) if dtype in (pl.Int64, pl.Int32, pl.Float64, pl.Float32)]
for col in numeric_cols:
    stats = descriptive_analytics(df[col].to_list())
    results[col] = stats


In [ ]:
# Prepare data for DataFrame
stats_rows = []
for col, stats in results.items():
    row = {"column": col}
    row.update(stats)
    stats_rows.append(row)

stats_df = pl.DataFrame(stats_rows)
stats_df

column,mean,median,mode,variance,std_dev,range
str,f64,f64,list[f64],f64,f64,f64
"""age""",15.924722,16.0,[13.0],4.086794,2.021582,16.0
"""daily_social_media_hours""",4.478785,4.4,[3.5],4.072426,2.018025,4.5
"""sleep_hours""",6.494183,6.5,"[6.8, 7.4]",2.050668,1.432015,6.5
"""screen_time_before_sleep""",1.742258,1.8,[1.4],0.511843,0.715432,1.75
"""academic_performance""",2.990248,2.99,"[2.36, 3.9]",0.333279,0.577303,3.0
"""physical_activity""",1.016168,1.0,[1.9],0.33949,0.582658,1.0
"""stress_level""",5.365269,5.0,[4.0],8.363895,2.89204,5.5
"""anxiety_level""",5.557742,6.0,[6.0],8.123589,2.850191,5.5
"""addiction_level""",5.571429,6.0,[6.0],7.988258,2.826351,5.5


# Outcome 
- **Mean Social Media Usage**  = **4.5**
- **Standard Deviation of Stress** = **2.89**

## Distribuiton Analysis